### PR TITLE
WSO-552 - Implement Auto Complete on the Graduate Explore Programs Page

### PR DESCRIPTION
**Pull request by dpfitzsi**

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_autocomplete/js/ucgy_core_autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        clearTimeout(this.timer);
+      }
+      this.timer = setTimeout(function() {
+        db.owner.setStatus('begin');
+
+        // Ajax GET request for autocompletion. We use Drupal.encodePath instead of
+        // encodeURIComponent to allow autocomplete search terms to contain slashes.
+        $.ajax({
+          type: 'GET', url: db.uri + '/' + Drupal.encodePath(searchString), dataType: 'json', success: function(matches) {
+            if (typeof matches.status == 'undefined' || matches.status != 0) {
```
Comment (SandyVallee) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_autocomplete/js/ucgy_core_autocomplete.js]: are we doing === in js?

File: private/migrations/2017-10-04_wso-489.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+set -e -u
+
+if [ -z "$PANTHEON_DEPLOY_ENV" ]; then
+  DRUSH_CMD="drush"
+else
+  DRUSH_CMD="terminus remote:drush $PANTHEON_DEPLOY_ENV -- "
+fi
+# enable the new module
+$DRUSH_CMD en ucgy_core_autocomplete --yes
```
Comment (taeram) [inline private/migrations/2017-10-04_wso-489.sh]: Should be converted to new ucgy-migrations format

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_autocomplete/js/ucgy_core_autocomplete.js (hunk 1 above)
Comment (taeram) [reply sites/all/modules/ucalgary/ucgy_core/ucgy_core_autocomplete/js/ucgy_core_autocomplete.js]: @SandyVallee @dpfitzsi MDN says we should be using `===`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Examples

**Comment by taeram** [comment]
Related to WSO-489.

Closing this PR, but keeping the branch for future development.

**Pull request closed without merging**